src/coap_client_libcoap.c

```diff
@@ -1179,6 +1179,77 @@ struct golioth_client *golioth_client_create(const struct golioth_client_config
     return NULL;
 }
 
+static void purge_request_mbox(golioth_mbox_t request_mbox)
+{
+    golioth_coap_request_msg_t request_msg = {};
+    size_t num_messages = golioth_mbox_num_messages(request_mbox);
+
+    for (size_t i = 0; i < num_messages; i++)
+    {
+        assert(golioth_mbox_recv(request_mbox, &request_msg, 0));
```

Comment:
I know this is copied code, but we shouldn't be wrapping function calls in `assert()`. Otherwise we could drop this call completely if asserts are turned off.
```suggestion
        bool ok = golioth_mbox_recv(request_mbox, &request_msg, 0);
        assert(ok);
```